File: ATG/app/src/main/java/com/kuhakupixel/atg/backend/ProcInfo.java
package com.kuhakupixel.atg.backend;

/*
 * info about process
 * */
public class ProcInfo {
    private String pidStr = "";
    private String name = "";

    /*
     * [pidAndNameStr]: pid and name separated by space
     *
     * */
    public ProcInfo(String pidAndNameStr) {
        String[] splitted = pidAndNameStr.split(" ", 2);
        assert (2 == splitted.length);
        this.pidStr = splitted[0];
        this.name = splitted[1];

    }

    public String GetPidStr() {
        return pidStr;
    }

    public String GetName() {
        return name;
    }
}

File: ATG/app/src/main/java/com/kuhakupixel/atg/backend/ProcInfo.kt
package com.kuhakupixel.atg.backend

/*
 * info about process
 * */
class ProcInfo(pidAndNameStr: String) {
    private var pidStr = ""
    private var name = ""

    /*
     * [pidAndNameStr]: pid and name separated by space
     *
     * */
    init {
        val splitted: List<String> = pidAndNameStr.split(" ", limit = 2)
        assert(2 == splitted.size)
        pidStr = splitted[0]
        name = splitted[1]
    }

    fun GetPidStr(): String {
        return pidStr
    }

    fun GetName(): String {
        return name
    }
}